File: sdk/mobile/android/src/main/kotlin/com/summit/odps/android/Consent.kt
package com.summit.odps.android

/**
 * Tracks the consent state for telemetry collection.
 */
enum class ConsentState {
  GRANTED,
  DENIED,
  UNKNOWN
}

class ConsentManager(initialState: ConsentState = ConsentState.UNKNOWN) {
  @Volatile
  private var state: ConsentState = initialState

  fun updateConsent(newState: ConsentState) {
    state = newState
  }

  fun hasConsent(): Boolean = state == ConsentState.GRANTED
}

File: sdk/mobile/ios/Sources/ODPS/Consent.swift
public enum ConsentState {
    case granted
    case denied
    case unknown
}

public final class ConsentManager {
    private var state: ConsentState

    public init(initialState: ConsentState = .unknown) {
        self.state = initialState
    }

    public func update(_ newState: ConsentState) {
        state = newState
    }

    public func hasConsent() -> Bool {
        state == .granted
    }
}
